test/common/upstream/health_checker_impl_test.cc

```diff
@@ -4487,6 +4487,60 @@ TEST_F(GrpcHealthCheckerImplTest, DontReuseConnectionBetweenChecks) {
   expectHostHealthy(true);
 }
 
+TEST_F(GrpcHealthCheckerImplTest, DontReuseConnectionTimeout) {
```

Comment:
Could you add a comment to this test case to make it easier to understand which assertion is verifying that we're not reusing the connection? I think its `expectClientCreate` but it would be nice with a comment alongside it